src/UsersModel.cpp

```diff
@@ -9,17 +9,37 @@
 #include <QUrl>
 
 #include "Cache.h"
+#include "Cache_p.h"
 #include "CompletionModelRoles.h"
 #include "UserSettingsPage.h"
 
 UsersModel::UsersModel(const std::string &roomId, QObject *parent)
   : QAbstractListModel(parent)
   , room_id(roomId)
 {
-    roomMembers_ = cache::roomMembers(roomId);
-    for (const auto &m : roomMembers_) {
-        displayNames.push_back(QString::fromStdString(cache::displayName(room_id, m)));
-        userids.push_back(QString::fromStdString(m));
+    // obviously, "friends" isn't a room, but I felt this was the least invasive way
+    if (roomId == "friends") {
+        auto e = cache::client()->getAccountData(mtx::events::EventType::Direct);
+        if (e) {
+            if (auto event =
+                  std::get_if<mtx::events::AccountDataEvent<mtx::events::account_data::Direct>>(
+                    &e.value())) {
+                for (const auto &[userId, roomIds] : event->content.user_to_rooms) {
+                    displayNames.push_back(
+                      QString::fromStdString(cache::displayName(roomIds[0], userId)));
+                    userids.push_back(QString::fromStdString(userId));
+                    avatarUrls.push_back(cache::avatarUrl(QString::fromStdString(roomIds[0]),
+                                                          QString::fromStdString(userId)));
+                }
+            }
+        }
+    } else {
+        for (const auto &m : cache::roomMembers(roomId)) {
+            displayNames.push_back(QString::fromStdString(cache::displayName(room_id, m)));
+            userids.push_back(QString::fromStdString(m));
+            avatarUrls.push_back(
+              cache::avatarUrl(QString::fromStdString(room_id), QString::fromStdString(m)));
```

Comment:
This actually adds quite some cost in some rooms. It is fine for now, but remember me to optimize this, after this is merged :3

Reply:
I was thinking this might be an issue, but I also thought that not using the avatarUrl for the friends "room" is adding unnecessary cache accesses. I didn't want to add additional logic for distinguishing the two and since the same process is used to get the displayNames, I thought the performance his is probably acceptable.